record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java

```diff
@@ -66,9 +69,28 @@ class InternalRecordBuilderProcessor {
         this.metaData = metaData;
         recordClassType = ElementUtils.getClassType(record, record.getTypeParameters());
         packageName = packageNameOpt.orElse(recordActualPackage);
-        builderClassType = ElementUtils.getClassType(packageName,
-                getBuilderName(record, metaData, recordClassType, metaData.suffix()), record.getTypeParameters());
-        typeVariables = record.getTypeParameters().stream().map(TypeVariableName::get).collect(Collectors.toList());
+
+        String builderName = getBuilderName(record, metaData, recordClassType, metaData.suffix());
+        recordTypeVariables = toTypeVariableNames(record.getTypeParameters());
+        List<TypeVariableName> builderTypeParams = new ArrayList<>(recordTypeVariables);
```

Comment:
The construction of `builderTypeParams` and `builderTypeParamsWildcard` shouldn't result in a modifiable `ArrayList`. Please move to separate method or `ElementUtils` so that the constructed type is immutable.